pages/api/web-scraping/flights.js

```diff
@@ -0,0 +1,187 @@
+// @ts-check
+import { FingerprintJsServerApiClient, Region } from '@fingerprintjs/fingerprintjs-pro-server-api';
+import { CheckResult, checkResultType } from '../../../server/checkResult';
+import { isRequestIdFormatValid, originIsAllowed, visitIpMatchesRequestIp } from '../../../server/checks';
+import { ALLOWED_REQUEST_TIMESTAMP_DIFF_MS, SERVER_API_KEY } from '../../../server/const';
+import { sendErrorResponse, sendForbiddenResponse, sendOkResponse } from '../../../server/response';
+import { ensurePostRequest, messageSeverity } from '../../../server/server';
+import { DAY_MS, FIVE_MINUTES_MS, HOUR_MS } from '../../../shared/const';
+import { AIRPORTS } from '../../web-scraping';
+
+/**
+ * @typedef {Object} ResultsQuery
+ * @property {string} from
+ * @property {string} to
+ * @property {string} requestId
+ * */
+
+/**
+ * @param {import('next').NextApiRequest} req
+ * @param {import('next').NextApiResponse} res
+ */
+export default async function getFlights(req, res) {
+  // This API route accepts only POST requests.
+  if (!ensurePostRequest(req, res)) {
+    return;
+  }
+
+  const { from, to, requestId } = /** @type {ResultsQuery} */ (req.body);
+
+  // Validate request ID format
+  if (!isRequestIdFormatValid(requestId)) {
+    sendForbiddenResponse(
+      res,
+      new CheckResult('Invalid request ID.', messageSeverity.Error, checkResultType.RequestIdMismatch)
+    );
+    return;
+  }
+
+  try {
```

Comment:
Do we need so big try-catch block?

Reply:
Yeah, it's large for my taste too, how would you restructure it? It's there so that if an unexpected error occurs, the server returns a 500 with a nice error message. Without the try-catch block, the server returns a 500 with a Next error HTML page in the body. 

I guess we could let next do that and limit the try-catch block to just calling the Server API. Then we can check for 500 response code on the client, throw away the HTML body, and just display our own non-specific "Server error" message.
Would that be preferable? Any better ideas?

Reply:
Yes, let's limit it only to the Server API call!

Reply:
Done :)